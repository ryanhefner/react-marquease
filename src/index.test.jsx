import React, { useRef } from 'react'
import { render } from '@testing-library/react'
import { configMocks, mockAnimationsApi } from 'jsdom-testing-mocks'
import {
  afterAll,
  afterEach,
  beforeAll,
  describe,
  expect,
  it,
  vi,
} from 'vitest'
import Marquee from './index'

configMocks({ afterAll, afterEach })

describe('react-marquease', () => {
  mockAnimationsApi()

  beforeAll(() => {
    window.innerWidth = 1024
  })

  it('renders', async () => {
    const { asFragment } = render(<Marquee>Test</Marquee>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders 2 clones', async () => {
    const { getAllByText } = render(<Marquee>Test</Marquee>)

    await getAllByText('Test')
    expect(getAllByText('Test')).toHaveLength(6)
  })
})
