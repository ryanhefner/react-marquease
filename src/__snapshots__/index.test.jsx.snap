// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`react-marquease > renders 1`] = `
<DocumentFragment>
  <div
    style="position: relative; overflow: clip; height: 0px;"
  >
    <div
      style="position: absolute; top: -9999px; left: -9999px;"
    >
      Test
    </div>
    <div
      aria-hidden="true"
      data-qa="scroller"
    >
      <div
        data-marquee-child="0"
        style="position: absolute; top: 0px; transform: translateX(0px);"
      >
        Test
      </div>
      <div
        data-marquee-child="1"
        style="position: absolute; top: 0px; transform: translateX(0px);"
      >
        Test
      </div>
    </div>
  </div>
</DocumentFragment>
`;
